### PR TITLE
VACMS-16499: Proper handling of image styles

### PR DESCRIPTION
**Pull request by ryguyk**
## Description
Relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16499

This PR cleans up the handling of image styles. It's a refactor in that functionality of the end product is not changed, but it's a bit more than that in that it changes the way in which developers interact with the component. 

There are two important pieces to note:

1. The biggest change is that developers now must pass the desired image style to the `MediaImage` component when using it. 
   ```
   <MediaImage
     className="event-detail-img"
     {...image}
     imageStyle="2_1_large"
   />
   ```
   Although we say "must", it's really something that allows for flexibility because we are no longer choosing the image style at the formatter level. Instead, we now pass to the component the entire collection of image styles, which allows for flexibility when using the image in the templates. Most notably, as an example, we could now use two different image styles, say, for two different screen sizes.

2. From Drupal, we only receive the height and width of _the original image_. We do not receive these dimensions for the individual image derivatives. This prevents us from being able to define a proper height and width on the `` tag, which creates a situation where [Cumulative Layout Shift](https://web.dev/articles/cls) is likely. So, as a result, this PR introduces some lightweight configuration for defining the height and width for the chosen image style. Ideally we'd get this information from Drupal, and perhaps there's a way, but that would be a separate ticket (maybe in this repo, maybe on the Drupal side, or maybe on both). It's important to note that this config (as well as the image-style naming) is _kind of_ coupled to Drupal. We are using image style names that are defined in Drupal, and we are setting this config based on the processing of the styles as it is defined in Drupal, BUT it's very arguable that these things would be named/configured exactly the same on another CMS. For example, there's nothing inherently "Drupal" about the name `2_1_large`, nor is there anything specific to Drupal about scaling that derivative to a specific size.

## Testing done
Manual

## Screenshots
image

Before image is loaded, it has proper 2:1 dimensions, thus preventing layout shift when it is loaded:
image

## QA steps
What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?

```[tasklist]
- [ ] Automated tests have passed
- [ ] Tugboat environment was generated without errors
```

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:

### CODE REVIEW

**Review by tjheffner — CHANGES_REQUESTED** [review]
overall, code changes look nice and seem a bit clearer as to what is actually happening with images + image styles.

most of the storybook images are now broken, however. specifically:
- MediaImage
- StoryListing
- NewsStory
- PersonProfile
- StaffProfile

File: src/data/queries/newsStory.ts [hunk 1]
```diff
@@ -45,10 +45,15 @@ export const formatter: QueryFormatter<NodeNewsStory, NewsStory> = (
...
-    }),
+    id: entity.id,
+    entityId: entity.drupal_internal__nid,
+    entityPath: entity.path.alias,
+    type: entity.type,
+    published: entity.status,
+    moderationState: entity.moderation_state,
+    title: entity.title,
+    metatags: entity.metatag,
+    image: queries.formatData('media--image', entity.field_media),
```
[the remark this answers is not in the document]
Comment (ryguyk) [reply src/data/queries/newsStory.ts:56]: I think this was a product of a merge conflict. This makes sense.

**Comment by ryguyk** [comment]
> most of the storybook images are now broken, however. specifically:
> 
> * MediaImage
> * StoryListing
> * NewsStory
> * PersonProfile
> * StaffProfile

@tjheffner Thanks. This is now resolved.

**Review by tjheffner — APPROVED** [review]
Thank you for this!

**Pull request merged**